darts/utils/data/tabularization.py

```diff
@@ -161,3 +163,390 @@ def _create_lagged_data(
+    `max_samples=1`.
+
+    Parameters
+    ----------
+    target_series
+        Optionally, the target series for the regression model to predict. This series only needs to be specified
+        if the trained lagged-feature model uses auto-regressive features.
+    past_series
+        Optionally, the past covariates series that the regression model will use as inputs.
+    future_series
+        Optionally, the future covariates (i.e. exogenous covariates) series that the regression model will use
+        as inputs.
+    target_lags
+        Optionally, the lags of the target series to be used as (auto-regressive) features.
+    lags_past_covariates
+        Optionally, the lags of `past_series` to be used as features.
+    lags_future_covariates
+        Optionally, the lags of `future_series` to be used as features.
+    lags_past_covariates
+        Optionally, the lags of `past_series` to be used as features.
+    lags_future_covariates
+        Optionally, the lags of `future_series` to be used as features.
+    max_samples
+        Optionally, the maximum number of samples to be drawn for training/validation; only the most recent samples
+        are kept. In theory, specifying a smaller `max_samples` should reduce computation time, especially in
+        cases where many observations could be generated.
+
+    Returns
+    -------
+    X
+        A `(n_observations, n_lagged_features, n_samples)`-shaped array of lagged features for training a
+        lagged-variables model.
+    times
+        The `time_index` of each observation in `X`.
+
+    Raises
+    ------
+    ValueError
+        If there are no times shared by all of the specified series, or if no lags have been specified.
+    TypeError
+        If the provided series do not share the same type of `time_index` (e.g. `target_series` uses a
+        `pandas.RangeIndex`, but `future_series` uses a `pandas.DatetimeIndex`).
+
+    """
+
+    all_series = [target_series, past_series, future_series]
+    all_lags = [target_lags, past_lags, future_lags]
+    _check_lags(all_lags)
+    times, times_idxs = _find_intersecting_times(
+        all_series, all_lags, max_samples, horizon=0, is_training=False
+    )
+    X = []
+    for i, time_idx in enumerate(times_idxs):
+        if time_idx is not None:
+            # Offset shared time indices by requested lags w/ + 1
+            # to account for fact that value at time `t` can also
+            # be used as a feature when predicting:
+            idx_to_get = time_idx.reshape(-1, 1) + all_lags[i] + 1
+            X.append(_create_lagged_vals(all_series[i], idx_to_get))
+    X = np.concatenate(X, axis=1)
+    return X, times
+
+
+def _check_lags(lags):
+    """
+    Raises `ValueError` if not at least one `*_lags` argument is specified.
+    """
+    raise_if(
+        all(lag is None for lag in lags),
+        ("Must specify at least one of: " "`target_lags`, `past_lags`, `future_lags`."),
+    )
+    return None
+
```

Comment:
I think we can skip checking the lags since this check is already done by the regression model, which will complain if no lags are provided, if I am not mistaken